api/spec/services/spree/webhooks/subscribers/handle_request_spec.rb

```diff
@@ -39,15 +34,42 @@
         expect(Rails.logger).to have_received(with_log_level).with(log_msg)
       end
 
-      it 'creates a webhooks event with the process info' do
+      it 'updates the event record created previously with the missing data' do
         expect { subject.call }.to change {
-          Spree::Webhooks::Event.all.as_json(except: %i[created_at id preferences updated_at]).map(&:values)
-        }.from([]).to([[execution_time, event, log_msg, response_code.to_s, subscriber.id, success, url]])
+          Spree::Webhooks::Event.
+            find(event.id).
+            as_json(except: %i[created_at id preferences updated_at]).
+            values
+        }.from(
+          [nil, event_name, nil, nil, subscriber.id, nil, url]
+        ).to(
+          [execution_time, event_name, log_msg, response_code.to_s, subscriber.id, success, url]
+        )
       end
 
       it { expect(subject.call).to eq(nil) }
     end
 
+    before do
+      expect(Spree::Webhooks::Event).to(
+        receive(:create).
+          with(name: event_name, subscriber_id: subscriber.id, url: url).
+          and_return(event)
+      )
+      expect(Spree::Webhooks::Subscribers::MakeRequest).to(
+        receive(:new).with(body: body.to_json, url: url).and_return(make_request_double)
```

Comment:
@damianlegawiec this is the only place where the body with the added event data is tested now. [Here](https://github.com/spree/spree/pull/11504/files#diff-726ef97c3571e54a204da8dfb105bf2ee55b61239453fc5b434311eb479eda9cR7) the event data is merged into the serializer hash for testing and [here](https://github.com/spree/spree/pull/11504/files#diff-d9f9039e1e67067424870ce8f4c2a70723bd047ac436880544b7755a714f6986R53) it's done right before making the webhook request.

Reply:
@sebastian-palma we need something more reliable than mocks here - we need a full test:

1. Order is placed
2. Job is created
3. Job is executed and we chang the payload being sent

So full E2E